integration-tests/combined-extension/lib.rs

```diff
@@ -0,0 +1,209 @@
+#![cfg_attr(not(feature = "std"), no_std, no_main)]
+
+use ink::{
+    env::{
+        DefaultEnvironment,
+        Environment,
+    },
+    ChainExtensionInstance,
+};
+use psp22_extension::Psp22Extension;
+use rand_extension::{
+    FetchRandom,
+    RandomReadErr,
+};
+
+/// This extension combines the [`FetchRandom`] and [`Psp22Extension`] extensions.
+/// It implements the [`ChainExtensionInstance`] trait by simply calling
+/// the corresponding sub-extension implementation.
+/// It is possible to combine any number of extensions in this way.
+#[ink::scale_derive(TypeInfo)]
+pub struct CombinedChainExtension;
```

Comment:
For similar reason, I am not a fun of having two structs associated with chain extensions. Can't you use `CombinedInstance` instead?

Reply:
The factory type should implement `TypeInfo` without any information regarding private types generated by the `#[ink::chain_extension]` macro. For that, we must either manually implement the `TypeInfo` trait, or have a separate instance type. Another reason is that the type that implements the `ink::ChainExtensionInstance` trait is a factory. If the factory produces self instance, it leads to incorrect usage of the pattern.